src/common/dao/project/projectmember.go

```diff
@@ -169,3 +168,30 @@ func SearchMemberByName(projectID int64, entityName string) ([]*models.Member, e
 	_, err := o.Raw(sql, queryParam).QueryRows(&members)
 	return members, err
 }
+
+// ListRoles lists the roles of user for the specific project
+func ListRoles(user *models.User, projectID int64) ([]int, error) {
+	if user == nil {
+		return nil, nil
+	}
+	var params []interface{}
+	sql := `
+		select role
+			from project_member
+			where entity_type = 'u' and entity_id = ? and project_id = ? `
+	params = append(params, user.UserID, projectID)
+	if len(user.GroupIDs) > 0 {
+		sql += fmt.Sprintf(`union
+			select role
+			from project_member
+			where entity_type = 'g' and entity_id in ( %s ) and project_id = ? `, dao.ParamPlaceholderForIn(len(user.GroupIDs)))
+		params = append(params, user.GroupIDs)
+		params = append(params, projectID)
+	}
+	roles := []int{}
+	_, err := dao.GetOrmer().Raw(sql, params).QueryRows(&roles)
```

Comment:
I think `union` in SQL doesn't prevent duplicated rows, do we expect duplicated values in the returned role list?

Reply:
Union will remove duplicates. Union All does not.

Reply:
You are right, my bad.